bool flag
int32[] data
---
bool res
